### PR TITLE
[EC-223] Add FIMS AppService

### PR DESCRIPTION
**Pull request by Krusty93**
### List of changes

Add AppService and related plan

### Motivation and context

Creating new FIMS resources

### Type of changes

- [X] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [X] No

### Other information

---

### CODE REVIEW

File: infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  plan_type = "internal"
+  plan_name = "${var.product}-oidc-provider-plan"
+  sku_name  = "B3"
+
+  name                = "${var.product}-oidc-provider-app"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+
+  always_on        = true
+  node_version     = "18-lts"
```
Comment (lucacavallaro) [inline infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf:13]: if is available, we can switch to `20-lts`

File: infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  app_command_line = "npm run start"
```
Comment (lucacavallaro) [inline infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf:14]: we should avoid using `npm` to start our node projects, the startup command should be `node .` (is it the default one?)

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

File: infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf (hunk 2 above)
Comment (Krusty93) [reply infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf:14]: `npm run start` is set by our app service TF module, and the current FIMS app service uses it.
Do you think the module shall be updated too?

Ok for this app service, I'll update it

Comment (lucacavallaro) [reply infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf]: I think the module should be updated, but I don't know if this change will have impact on production.

Using `npm start` is a bad practice, but maybe our legacy projects rely on it.

File: infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf (hunk 1 above)
Comment (pagopa-engineering-github) [reply infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf]: it is, I've updated the PR

Comment (Krusty93) [reply infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf]: It does, I've updated it

**Review by lucacavallaro — APPROVED** [review]

File: infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf [hunk 3]
```diff
@@ -0,0 +1,34 @@
+module "appservice_fims" {
+  source = "github.com/pagopa/terraform-azurerm-v3//app_service?ref=v7.67.1"
+
+  plan_type = "internal"
+  plan_name = "${var.product}-oidc-provider-plan"
+  sku_name  = "B3"
+
+  name                = "${var.product}-oidc-provider-app"
```
Comment (lucacavallaro) [inline infra/prod/_modules/app_services/app_service_fims_oidc_provider.tf:8]: I was wrong! the right name is `openid-provider` ([reference](https://openid.net/specs/openid-connect-core-1_0.html#Terminology:~:text=an%20OpenID%20Provider.-,OpenID%20Provider%20(OP),-OAuth%202.0%20Authorization))

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

**Pull request merged**